### DISCUSSION TITLE
Installation in a docker on unraid

### DISCUSSION THREAD

**Discussion by pixbalance** [Part-DB/Part-DB-server#388] [General]
Hi,

I am trying to get part-db running on a docker on unraid. Maybe I get some advice. 

The Database is "/mnt/user/appdata/partdb/db" but I did not figure out how to set the datebase_url. The app.db is created in the folder but has 0B.

`
docker run
  -d
  --name='partdb'
  --net='br0'
  --ip='172.16.0.77'
  -e TZ="Europe/Berlin"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="TAB-HomeServer"
  -e HOST_CONTAINERNAME="partdb"
  -e 'TCP_PORT_80'='80'
  -e 'DATABASE_URL'='sqlite:///%kernel.project_dir%/partdb/db/app.db'
  -e 'APP_ENV'='docker'
  -e 'DEFAULT_LANG'='de'
  -e 'DEFAULT_TIMEZONE'='Europe/Berlin'
  -e 'BASE_CURRENCY'='EUR'
  -e 'INSTANCE_NAME'='Part-DB'
  -e 'ALLOW_ATTACHMENT_DOWNLOADS'='0'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='http://[IP]:[PORT:80]'
  -l net.unraid.docker.icon='https://github.com/pawelmalak/unraid-templates/blob/master/templates/icons/partdb.png?raw=true'
  -v '/mnt/cache/appdata/partdb/uploads':'/var/www/html/uploads':'rw'
  -v '/mnt/cache/appdata/partdb/media':'/var/www/html/public/media':'rw'
  -v '/mnt/user/appdata/partdb/db':'/var/www/html/var/db':'rw'
  --restart unless-stopped 'jbtronics/part-db1:latest'
da08308028a87770d5ea16dd9c09507108c46b232cba785ae24d49cbaf0459d9
`

**Comment by pixbalance** [comment]
btw, I get the error message:

Oops! An Error Occurred
The server returned a "500 Internal Server Error".
Something is broken. Please let us know what you were doing when this error occurred. We will fix it as soon as possible. Sorry for any inconvenience caused.

**Reply by w0hc** [reply]
I have (had) it running on Unraid, but my motherboard appears to have died so I'm not able to check right now how I set it up. I am using sqlite rather than an external database, if that helps. Hope to have my new server parts here in a few days to rebuild then I can probably help more.

I get the error 500 quite often, usually when trying to import parts from Digikey.

**Reply by w0hc** [reply]
Also, I assume you are installing it from Community Apps and not manually?

**Reply by pixbalance** [reply]
correct. I used the community apps. I assume I still put in the wrong path to the sqlite database...

**Comment by w0hc** [comment]
I was able to get my box running. Your path is correct, and my database url is:
`sqlite:///%kernel.project_dir%/var/db/app.db`
Give that a try.

**Reply by pixbalance** [reply]
if I use 
-v '/mnt/user/appdata/partdb/db':'/var/www/html/var/db':'rw

and
sqlite:///%kernel.project_dir%/var/db/app.db

i get another error:
![image](https://github.com/Part-DB/Part-DB-server/assets/62733848/5a40d18c-9dfe-488b-b319-6c44f77b25a2)

**Reply by w0hc** [reply]
Here's my database settings from the docker edit page:
`Database: /mnt/user/appdata/partdb/db`

`DATABASE_URL: sqlite:///%kernel.project_dir%/var/db/app.db`

**Reply by pixbalance** [reply]
tried several times and reinstall without success. Thanks.

**Reply by w0hc** [reply]
Be sure to read the docs here: https://docs.part-db.de/installation/installation_docker.html

Have you ran
`docker exec --user=www-data partdb php bin/console doctrine:migrations:migrate
`
from the Unraid console?

**Reply by pixbalance** [reply]
I totally missed that. run the line and it is working. :/ THX
